src/ripple/app/tx/impl/InvariantCheck.cpp

```diff
@@ -775,18 +778,28 @@ ValidClawback::finalize(
             return false;
         }
 
-        AccountID const issuer = tx.getAccountID(sfAccount);
-        STAmount const amount = tx.getFieldAmount(sfAmount);
-        AccountID const& holder = amount.getIssuer();
-        STAmount const holderBalance = accountHolds(
-            view, holder, amount.getCurrency(), issuer, fhIGNORE_FREEZE, j);
-
-        if (holderBalance.signum() < 0)
+        if (mptokensChanged > 1)
         {
             JLOG(j.fatal())
-                << "Invariant failed: trustline balance is negative";
+                << "Invariant failed: more than one mptokens changed.";
             return false;
         }
+
+        if (trustlinesChanged == 1)
+        {
+            AccountID const issuer = tx.getAccountID(sfAccount);
+            STAmount const amount = tx.getFieldAmount(sfAmount);
```

Comment:
Should be ref.

Reply:
Should the issuer be a ref as well? `getAccountID` returns a temporary value, not sure if it's safe to assign to a const ref.